xarray/core/common.py

```diff
@@ -479,8 +482,38 @@ def where(self, cond):
         Coordinates:
           * y        (y) int64 0 1 2 3 4
           * x        (x) int64 0 1 2 3 4
+        >>> a.where((a > 6) & (a < 18), drop=True)
+        <xarray.DataArray (x: 5, y: 5)>
+        array([[ nan,  nan,   7.,   8.,   9.],
+               [ 10.,  11.,  12.,  13.,  14.],
+               [ 15.,  16.,  17.,  nan,  nan],
+        Coordinates:
+          * x        (x) int64 1 2 3
+          * y        (y) int64 0 1 2 3 4
         """
-        return self._where(cond)
+        if drop:
```

Comment:
Indeed this is an arg that could be kwarg-only in python3. (but we can't do that here)